src/akamai/js/akamaiHD_flash.js

```diff
@@ -0,0 +1,1334 @@
+/*
+ * Akamai HD flash video plugin
+ */
+
+require("../../../html5-common/js/utils/InitModules/InitOO.js");
+require("../../../html5-common/js/utils/InitModules/InitOOUnderscore.js");
+require("../../../html5-common/js/utils/InitModules/InitOOHazmat.js");
+require("../../../html5-common/js/utils/constants.js");
+(function(_, $) {
+  var pluginName = "ooyalaFlashVideoTech";
+  var flashMinimumVersion = "11.1.0";
+  var cssFromContainer;
+
+  /**
+   * Config variables for paths to flash resources.
+   */
+
+  var pluginPath;
+  var filename = "akamaiHD_flash.*\.js";
+  var scripts = document.getElementsByTagName('script');
+  for (var index in scripts) {
+    var match = scripts[index].src.match(filename);
+    if (match && match.length > 0) {
+      pluginPath = match.input.match(/.*\//)[0];
+      break;
+    }
+  }
+  if (!pluginPath) {
+    console.error("Can't get path to script", filename);
+    return;
+  }
+  pluginPath += "akamaiHD_flash.swf";
+  var flexPath = "playerProductInstall.swf";
+  this.ready = false;
+  /**
+   * @class OoyalaFlashVideoFactory
+   * @classdesc Factory for creating video player objects that use Flash in an HTML5 wrapper.
+   * @property {string} name The name of the plugin
+   * @property {boolean} ready The readiness of the plugin for use.  True if elements can be created.
+   * @property {object} encodings An array of supported encoding types (ex. m3u8, mp4)
+   */
+
+  var OoyalaFlashVideoFactory = function() {
+    this.name = pluginName;
+    // This module defaults to ready because no setup or external loading is required
+    this.ready = true;
+    /**
+     * Checks whether flash player is available
+     * @public
+     * @method getFlashVersion
+     * @returns encoding as hds if flash version is available
+     */
+    function getFlashVersion() {
```

Comment:
Since some of these things are common between the osmf plugins i would recommend either moving them to a common file that gets built into both plugins or moving them to the html5-common repo.  Not required, just suggesting.  Bitmovin will need the flash logic soon as well